includes/admin/class-wc-admin-post-types.php

```diff
@@ -986,6 +868,107 @@ public function add_display_post_states( $post_states, $post ) {
+	 * @param string     $stock_status The stock status to use for adjustment.
+	 * @return WC_Product The supplied product, or the synced product if it was a variable product.
+	 */
+	private function update_stock_status( $product, $stock_status ) {
+		if ( $product->is_type( 'external' ) ) {
+			// External products are always in stock.
+			$product->set_stock_status( 'instock' );
+		} elseif ( $product->is_type( 'variable' ) && ! $product->get_manage_stock() ) {
+			// Stock status is determined by children.
+			foreach ( $product->get_children() as $child_id ) {
+				$child = wc_get_product( $child_id );
+				if ( ! $product->get_manage_stock() ) {
+					$child->set_stock_status( $stock_status );
+					$child->save();
+				}
+			}
+			$product = WC_Product_Variable::sync( $product, false );
+		} else {
+			$product->set_stock_status( $stock_status );
+		}
+
+		return $product;
+	}
+
+	/**
+	 * Set the new regular or sale price if requested.
+	 *
+	 * @param WC_Product $product The product to set the new price for.
+	 * @param string     $price_type 'regular' or 'sale'.
+	 * @return bool true if a new price has been set, false otherwise.
+	 */
+	private function set_new_price( $product, $price_type ) {
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+
+		if ( empty( $_REQUEST[ "change_{$price_type}_price" ] ) || ! isset( $_REQUEST[ "_{$price_type}_price" ] ) ) {
+			return false;
+		}
+
+		$old_price     = $product->{"get_{$price_type}_price"}();
+		$price_changed = false;
+
+		$change_price  = absint( $_REQUEST[ "change_{$price_type}_price" ] );
+		$raw_price     = wc_clean( wp_unslash( $_REQUEST[ "_{$price_type}_price" ] ) );
+		$is_percentage = (bool) strstr( $raw_price, '%' );
+		$price         = wc_format_decimal( $raw_price );
+
+		switch ( $change_price ) {
+			case 1:
+				$new_price = $price;
+				break;
+			case 2:
+				if ( $is_percentage ) {
+					$percent   = $price / 100;
+					$new_price = $old_price + ( round( $old_price * $percent, wc_get_price_decimals() ) );
+				} else {
+					$new_price = $old_price + $price;
+				}
+				break;
+			case 3:
+				if ( $is_percentage ) {
+					$percent   = $price / 100;
+					$new_price = max( 0, $old_price - ( round( $old_price * $percent, wc_get_price_decimals() ) ) );
+				} else {
+					$new_price = max( 0, $old_price - $price );
+				}
+				break;
+			case 4:
+				if ( 'sale' !== $price_type ) {
+					break;
+				}
+				if ( $is_percentage ) {
+					$percent   = $price / 100;
+					$new_price = max( 0, $product->regular_price - ( round( $product->regular_price * $percent, wc_get_price_decimals() ) ) );
+				} else {
+					$new_price = max( 0, $product->regular_price - $price );
+				}
+				break;
+
+			default:
+				break;
+		}
+
+		if ( isset( $new_price ) && $new_price !== $old_price ) {
+			$price_changed = true;
+			if ( 'sale' === $price_type && empty( $new_price ) && '0' !== $new_price ) {
+				$new_price = '';
```

Comment:
I've kept this logic from the pre-refactored code, but I'm not sure if it's really necessary; I can't think of any case where the price would end up being `''` (the previous `isset` check makes the `empty` check useless, right?)

Reply:
I agree that it does not look necessary. maybe remove it?